src/lightcurvelynx/models/physical_model.py

```diff
@@ -371,51 +414,102 @@ def compute_sed_with_extrapolation(self, times, wavelengths, graph_state, **kwar
+
+        # Get the flux density at all times and the valid wavelengths. Place these in a zero-padded
+        # array, trimming the first and last values along each axis since they were added to ensure
+        # the flux is computed at the min and max wavelengths and times.
+        computed_flux = self.compute_sed(query_times, query_waves, graph_state)
+        flux_density_all_waves = np.zeros((len(query_times), len(wavelengths)))
+        flux_density_all_waves[:, in_wave_range] = computed_flux[:, 1:-1]
+
+        # We do the extrapolation in two steps: first for wavelengths and then for times. This is
+        # because we extrapolate the fluxes at all wavelengths (included the extrapolated ones) to
+        # each new time. The result is that we combine the extrapolation for both dimensions.
+        if self.wave_extrapolation is not None and not np.all(in_wave_range):
             # Compute the flux values before the model's first valid wavelength.
-            flux_density[:, before_mask] = self.wave_extrapolation(
+            flux_density_all_waves[:, before_wave_mask] = self.wave_extrapolation.extrapolate_wavelength(
                 min_valid_wave,
                 computed_flux[:, 0],
-                wavelengths[before_mask],
+                wavelengths[before_wave_mask],
             )
 
             # Compute the flux values after the model's last valid wavelength.
-            flux_density[:, after_mask] = self.wave_extrapolation(
+            flux_density_all_waves[:, after_wave_mask] = self.wave_extrapolation.extrapolate_wavelength(
                 max_valid_wave,
                 computed_flux[:, -1],
-                wavelengths[after_mask],
+                wavelengths[after_wave_mask],
             )
 
+        # Then do extrapolation for times that fell outside the model's bounds.
+        if not np.all(in_time_range):
+            flux_density = np.zeros((len(times), len(wavelengths)))
+            flux_density[in_time_range, :] = flux_density_all_waves[1:-1, :]
+
+            if self.time_extrapolation is not None:
+                # Compute the flux values before the model's first valid time.
+                flux_density[before_time_mask, :] = self.time_extrapolation.extrapolate_time(
+                    min_valid_time,
+                    flux_density_all_waves[0, :],
+                    times[before_time_mask],
+                )
+
+                # Compute the flux values after the model's last valid time.
+                flux_density[after_time_mask, :] = self.time_extrapolation.extrapolate_time(
+                    max_valid_time,
+                    flux_density_all_waves[-1, :],
+                    times[after_time_mask],
+                )
+            else:
+                warnings.warn(
+                    "The time_extrapolation is None but some times are outside the model's valid range. "
+                    "These values will be set to zero."
```

Comment:
Some more debugging information may be useful. Maybe times you are talking about, or even model parameters?

Reply:
I was originally thinking this, but the problem with debugging information is that I think it would issue a unique warning for each set of output instead of a single one.